### PR TITLE
Another step: New sinkholing logic, logging.

### PR DESCRIPTION
**Pull request by Karm**

### CODE REVIEW

**Comment by Karm** [comment]
Do not merge -- investigating the failure.

**Comment by Karm** [comment]
@kozeltom Fixed. You can evaluate and merge.

File: ejb/src/main/java/biz/karms/sinkit/eventlog/EventLogFactory.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+public class EventLogFactory {
+    public static EventLogRecord getEventLogRecord(EventLogAction action, String clientIPAddress, String[] matchedIoCs, boolean isFQDN, String fqdnOrIp, String documentId) {
+        EventLogRecord eventLogRecord = new EventLogRecord();
+        eventLogRecord.setAction(action);
+        eventLogRecord.setClient(clientIPAddress);
+        eventLogRecord.setDocumentId(documentId);
+        eventLogRecord.setLogged(Calendar.getInstance().getTime());
+        // Why Array? It's always gonna be 1 IoC
+        eventLogRecord.setMatchedIocs(matchedIoCs);
```
Comment (Karm) [inline ejb/src/main/java/biz/karms/sinkit/eventlog/EventLogFactory.java]: @kozeltom @robcza Why is this supposed to be an array?

File: ejb/src/main/java/biz/karms/sinkit/eventlog/EventLogFactory.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        if (isFQDN) {
+            eventReason.setFqdn(fqdnOrIp);
+        } else {
+            eventReason.setIp(fqdnOrIp);
+        }
+        eventLogRecord.setReason(eventReason);
+        EventDNSRequest eventDNSRequest = new EventDNSRequest();
+        // Why? What? How is this different from  eventLogRecord.setClient()? :-(
+        //eventDNSRequest.setIp();
+        //eventDNSRequest.setRaw();
```
Comment (Karm) [inline ejb/src/main/java/biz/karms/sinkit/eventlog/EventLogFactory.java]: @robcza @kozeltom What is this for?

File: ejb/src/main/java/biz/karms/sinkit/eventlog/EventLogFactory.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        eventLogRecord.setRequest(eventDNSRequest);
+        // No idea...
+        //eventLogRecord.setVirusTotalRequest();
```
Comment (Karm) [inline ejb/src/main/java/biz/karms/sinkit/eventlog/EventLogFactory.java]: @kozeltom @robcza What should I log here?

**Comment by kozeltom** [comment]
You should not access the ArchiveService directly. Access only CoreService which contains method for logging. You don't have to construct EventLogRecord manually, everything is done in CoreService.

**Comment by Karm** [comment]
@kozeltom I fixed the logging. Please, review.

@robcza Circle CI runs both PostgreSQL and ElasticSearch for the test with each pull request. I :heart: it.

**Pull request merged**